docs/reference/search/terms-enum.asciidoc

```diff
@@ -35,7 +35,7 @@ The API returns the following response:
 }
 --------------------------------------------------
 
-The "complete" flag is false if time or space constraints were met and the
+The "complete" flag is false if time or space constraints were not met and the
 set of terms examined was not the full set of available values.
```

Comment:
Thanks for raising this @amitmbm. I agree that the current wording in ambiguous. In particular, I'm not sure what "space" constraints would mean. It seems that this would most likely occur due to a timeout.

I've suggested some changes to clarify:
```suggestion
If the `complete` flag is `false`, the returned `terms` set may be incomplete.
This typically happens if the request times out.
```

However, I'd like to get @markharwood's review before merging these changes in.

Reply:
The flag is intended to cover any number of scenarios where we might want the user to "try anyway" i.e. keep on typing the word that is in their head and search for it even if they don't see the word offered in the completion suggestions list.

The reasons for a term not being in the results could be varied e.g. a timeout or an error on one of the nodes. In future we might want to sort by popularity rather than alphabetically and this will introduce some data-gathering size limits that could lead to missing data the same way `terms` aggs can be less than accurate today.

Rather than users having to check a multitude of flags and conditions the thinking was this would be one flag which would help say if the suggestions were exhaustive regardless of the details of why that may be so.

Reply:
Thanks @markharwood. I made some adjustments to the suggestion based on your feedback. My hope is that we can be a little more specific while still future-proofing this for the `terms`-agg-like size constraints you mentioned. Let me know what you think!